pkg/server/event_handler.go

```diff
@@ -293,6 +299,46 @@ func (h *EventHandler[T, P]) getGCSObjectProto(ctx context.Context, objAttrs map
 	return p, nil
 }
 
+func (h *EventHandler[T, P]) getGCSObjectGithubResourceMetadata(ctx context.Context, objAttrs map[string]string) (*v1alpha1.GitHubSource, error) {
+	bucketID, found := objAttrs["bucketId"]
+	if !found {
+		return nil, fmt.Errorf("bucket ID not found")
+	}
+	objectID, found := objAttrs["objectId"]
+	if !found {
+		return nil, fmt.Errorf("object ID not found")
+	}
+
+	// Read the object from bucket.
+	attrs, err := h.client.Bucket(bucketID).Object(objectID).Attrs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("object attrs not found")
+	}
```

Comment:
I don't think we need to call GCS again. The PubSub notification we got should already have the metadata. See: https://cloud.google.com/storage/docs/pubsub-notifications#payload